### PR TITLE
Update testComponents to use addComponent

### PR DESCRIPTION
**Pull request by aseth1**
testComponents is no longer limited to ModelComponents.  Now that any component can be added to Model, we do not need to limit to tested components to be of type `ModelComponent`. It now throws if the object being tested is not a `Component` and causes the test to fail.

### CODE REVIEW

File: OpenSim/Tests/Components/testComponents.cpp [hunk 1]
```diff
@@ -217,7 +217,8 @@ void testComponent(const Component& instanceToTest)
 
             if (dependency == nullptr){
                 // Get a concrete instance of a PhysicalFrame, which is a Body
-                if (dependencyTypeName == "PhysicalFrame"){
+                size_t n = dependencyTypeName.length();
+                if (n > 4 && dependencyTypeName.substr(n-5) == "Frame") {
```
Comment (chrisdembia) [inline OpenSim/Tests/Components/testComponents.cpp]: It'd be great if we could use `std::is_base_of` or casts instead of this.

**Comment by chrisdembia** [comment]
Done reviewing. Made 1 comment.

File: OpenSim/Tests/Components/testComponents.cpp (hunk 1 above)
Comment (aseth1) [inline OpenSim/Tests/Components/testComponents.cpp]: I agree, but here we do not even have an object, just the type name of the dependency.
Maybe there is a better way without using the registered types.

File: OpenSim/Tests/Components/testComponents.cpp [hunk 2]
```diff
@@ -467,40 +468,28 @@ void addObjectAsComponentToModel(Object* instance, Model& model)
...
-    // It is more than likely that connect() will fail, but the subcomponents tree
-    // will be traversable, so we can continue to resolve dependencies by visiting
-    // subcomponents' connectors
-    catch (const std::exception& e) {
-        cout << "testComponents: Model unable to connect after adding ";
-        cout << instance->getName() << endl;
-        cout << "ERROR: " << e.what() << "'" << endl;
-        cout << "Possible that dependency was not added yet. Continuing...." << endl;
+    if (Object::isObjectTypeDerivedFrom< Analysis >(className))
+        model.addAnalysis(dynamic_cast<Analysis*>(instance));
```
Comment (chrisdembia) [inline OpenSim/Tests/Components/testComponents.cpp]: Would be good to add a comment here. Connectors *can* connect to Objects but it is not tested and may not work (only components have paths).

File: OpenSim/Tests/Components/testComponents.cpp (hunk 1 above)
Comment (chrisdembia) [inline OpenSim/Tests/Components/testComponents.cpp]: You can do the same thing as below: `isObjectDerivedFrom<Frame>(dependencyTypeName)`

**Comment by chrisdembia** [comment]
I restarted some of the builds.

**Comment by tkuchida** [comment]
AppVeyor is likely failing because of MSVC Update 3 bug (see #1115).

**Comment by chrisdembia** [comment]
I restarted the appveyor build b/c we fixed the appveyor MSVC update issue in master.

**Comment by chrisdembia** [comment]
appveyor/pr now passes. Merging.

**Comment by aseth1** [comment]
Many thanks!

**Pull request merged**